lib/spack/spack/cmd/flake8.py

```diff
@@ -242,7 +246,7 @@ def setup_parser(subparser):
 
 
 def flake8(parser, args):
-    flake8 = which('flake8', required=True)
+    flake8 = get_executable('flake8', spec='py-flake8', install=True)
```

Comment:
We currently use `which` in a lot of places in Spack. All of those places could benefit from that functionality. Would it be possible to instead add a `install=True` arg to `which` much like `working_dir` has a `create=True` arg? Or would that lead to unsolvable circular import issues?

Reply:
I'm torn about moving this functionality into `util/executable.py`. It seems most useful there, but the goal with `util.*` is that they don't need to import from `spack.*`, and this definitely needs `spack.spec` and `spack.store`. I think this is something that should be discussed more before merging.

Reply:
One possibility here is to use something like https://github.com/pantsbuild/pex to separate the python environment for flake8 from the rest of spack. I had started on making a PexPackage base class which would have made this relatively easy, I will evaluate whether that would be useful here.

Reply:
I don't actually want the python environment separated (or at least I'm pretty sure I don't). I'm relying on the fact that we know of one working python on the system (namely, the one Spack is running in) to save huge amounts of build time. If we bootstrap flake8 without relying on the current python executable, then we have to pay the cost of building (or downloading a binary for) python and all of its dependencies. Why bother with that when we know where python is?

Reply:
Sorry, when I said "python environment" I was referring to the set of installed packages. If we were to generate any PEX files e.g. for flake8, we would definitely just point pex at the resolved python executable (pex emphatically will never try to build python, it will only search for interpreters).

The reason this would come in handy is if e.g. `flake8` specifies a package that conflicts with `coverage` or something. It would mean that we could create executables like `flake8` without modifying the python environment (and therefore the PEX file would stay cached).